### PR TITLE
Refactor vm apis to harvester

### PR DESCRIPTION
**Pull request by guangbochen**
Refactor the `vm` apis to `harvester`

Related Issue: https://github.com/rancher/harvester/issues/54

### CODE REVIEW

File: pkg/config/context.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ type Management struct {
 	ControllerFactory controller.SharedControllerFactory
 
 	VirtFactory    *kubevirt.Factory
-	VMFactory      *vm.Factory
+	HarvFactory    *harvester.Factory
```
Comment (gitlawr) [inline pkg/config/context.go]: I would prefer HarvesterFactory for the field

Comment (guangbochen) [reply pkg/config/context.go]: Done

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/config/context.go [hunk 2]
```diff
@@ -99,11 +99,11 @@ func setupManagement(ctx context.Context, restConfig *rest.Config, opts *generic
 	management.VirtFactory = virt
 	management.starters = append(management.starters, virt)
 
-	vm, err := vm.NewFactoryFromConfigWithOptions(restConfig, opts)
+	vm, err := harvester.NewFactoryFromConfigWithOptions(restConfig, opts)
```
Comment (orangedeng) [inline pkg/config/context.go]: Also rename `vm` variable

**Comment by orangedeng** [comment]
Need rebase.

File: pkg/config/context.go (hunk 2 above)
Comment (guangbochen) [reply pkg/config/context.go]: done

**Comment by guangbochen** [comment]
@orangedeng rebased thanks.

**Review by orangedeng — APPROVED** [review]
LGTM

**Pull request merged**